libraries/stdlib/genosl/lib/vector4_extra_ops.osl

```diff
@@ -0,0 +1,45 @@
+// Adds some syntactic sugar allowing mixing vector4 and color4 as
+// arguments of some binary operators used by OCIO transform code.
+
+vector4 __operator__mul__(matrix m, vector4 v)
+{
+  return vector4(v.x * m[0][0] + v.y * m[0][1] + v.z * m[0][2] + v.w * m[0][3], 
```

Comment:
I was going to suggest this wrap the existing `transform(m, v)` function that exists in `libraries/stdlib/genosl/lib/vector_4.h`, but then when I was looking it looks like the math isn't actually the same.

https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/libraries/stdlib/genosl/include/vector4.h#L414

I remember @jstone-lucasfilm mentioning something about extensive documentation about matrix vector multiplication in MaterialX but I can't find it right now - but I would like to propose that these two function at least agree with each other - or even better be a single shared implementation.

Reply:
Here's the documentation for matrix-vector multiplication in MaterialXCore, and our shading language implementations ought to follow this same pattern:

https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXCore/Types.h#L653